pkg/oc/cli/observe/observe.go

```diff
@@ -327,23 +329,58 @@ func (o *ObserveOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args [
 		return err
 	}
 
-	switch o.templateType {
-	case "jsonpath":
-		p, err := NewJSONPathArgumentPrinter(o.includeNamespace, o.strictTemplates, o.templates...)
-		if err != nil {
-			return err
+	// TODO: Remove in the next release
+	// support backwards compatibility with misspelling of "go-template" output format
+	if o.PrintFlags.OutputFormat != nil && *o.PrintFlags.OutputFormat == "gotemplate" {
+		fmt.Fprintf(o.ErrOut, "DEPRECATED: The %q output format has been replaced by %q, please use the new spelling instead\n", "gotemplate", "go-template")
+		*o.PrintFlags.OutputFormat = "go-template"
+	}
+
+	// TODO: Remove in the next release
+	// support backwards compatibility with incorrect flag --strict-templates
+	if o.strictTemplates {
+		fmt.Fprintf(o.ErrOut, "DEPRECATED: %q has been replaced by %q, please use the new flag instead\n", "--strict-templates", "--allow-missing-template-keys=false")
+		*o.PrintFlags.TemplatePrinterFlags.AllowMissingKeys = !o.strictTemplates
+	}
+
+	// prevent output-format from defaulting to go-template if no --output format
+	// is explicitly provided by the user
+	o.PrintFlags.OutputFlagSpecified = func() bool { return true }
+	printerList := []printers.ResourcePrinter{}
+
+	p, err := o.PrintFlags.ToPrinter()
+	if err == nil {
+		// here we only want to see if a valid output format containing a template arg was specified
+		// and there is a matching printer. If so, add that printer to our list
+		printerList = append(printerList, p)
+	}
+
+	// normalize --output value, keep the specified go-template or jsonpath output format,
+	// but remove any template arg specified as part of it
+	if o.PrintFlags.OutputFormat != nil {
```

Comment:
Oh my :sob:

Reply:
:) due to the nature of this command, it'll end up having to know a bit about the PrintFlags implementation